src/Language/Mulang/Inspector/Family.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE LiberalTypeSynonyms #-}
+
+module Language.Mulang.Inspector.Family (
+  deriveUses,
+  deriveDeclares,
+  InspectionFamily,
+  BoundInspectionFamily) where
+
+import Language.Mulang.Consult (Consult)
+import Language.Mulang.Counter (Count, Counter)
+import Language.Mulang.Inspector.Matcher (unmatching, Matcher)
+import Language.Mulang.Inspector.Primitive (countExpressions, positive, Inspection)
+import Language.Mulang.Inspector.Bound (countBoundDeclarations, uncounting, BoundConsult, BoundCounter, BoundInspection)
+
+type Family a = ((a Bool), Matcher -> (a Bool), Matcher -> (a Count))
+type InspectionFamily = Family Consult
+type BoundInspectionFamily = Family BoundConsult
+
+
+-- | Derives a family of usage inspections from a primal   inspection,
+-- which consist of a simple inspection, a matching inspection and a counter
+deriveUses :: (Matcher -> Inspection) -> InspectionFamily
+deriveUses f = (uses, usesMatching, countUses)
+  where
+    uses :: Inspection
+    uses = unmatching usesMatching
+
+    usesMatching :: Matcher -> Inspection
+    usesMatching matcher = positive (countUses matcher)
+
+    countUses :: Matcher -> Counter
+    countUses matcher = countExpressions (f matcher)
+
+-- | Derives a family of declarration inspections from a primal inspection,
```

Comment:
typo in declaration :stuck_out_tongue_closed_eyes:

Reply:
-.-